src/bun.js/bindings/bindings.cpp

```diff
@@ -167,6 +167,22 @@ static bool canPerformFastPropertyEnumerationForIterationBun(Structure* s)
     return true;
 }
 
+JSValue getIndexWithoutAccessors(JSGlobalObject* globalObject, JSObject* obj, uint64_t i)
+{
+    if (obj->canGetIndexQuickly(i)) {
+        return obj->tryGetIndexQuickly(i);
+    }
+
+    PropertySlot slot(obj, PropertySlot::InternalMethodType::Get);
+    if (obj->methodTable()->getOwnPropertySlotByIndex(obj, globalObject, i, slot)) {
+        if (!slot.isAccessor()) {
```

Comment:
are you sure we shouldn't call accessors here?

```js
❯ cat /tmp/a.js
       │ File: /tmp/a.js
   1   │ const obj = {};
   2   │ Object.defineProperty(obj, "0", {get() { return 42; }});
   3   │ console.log(obj[0])

❯ bun /tmp/a.js
42
```

perfectly valid, terrible code

Reply:
yes, this passes in jest:
```js
const obj1 = {};
Object.defineProperty(obj1, "0", {
  get() {
    return 42;
  },
});
const obj2 = {};
Object.defineProperty(obj2, "0", {
  get() {
    return 44;
  },
});
expect(obj1).toStrictEqual(obj2);
```

Reply:
ah, but maybe `deepEquals` outside of jest shouldn't behave this way

Reply:
ah, then lets wait for someone to file an issue before we add more complexity